---
title: "MS5"
format: 
  html:
    embed-resources: true
    theme: flatly
    toc: true
    toc-depth: 3
    toc-expand: 3
    toc-title: Contents
    smooth-scroll: true
editor: visual
params:
  organism: human
---

# Loading packages

```{r}
#| echo: false
#| include: false
#| warning: false

library(dplyr)
library(ggplot2)
library(tidyr)
library(readxl)
library(kableExtra)
library(ggpubr)
library(plotly)
library(car)  # for ANOVA
library(multcomp)  # for post-hoc tests
library(rcompanion)  # for wtd.boxplot.stats
#BiocManager::install("idpr")
#library(idpr)
```

# Loading dataframes

```{r}
#| echo: false
#| include: false

# This is just a filtered by Post-translational
human_ptms <- read.csv(file = "/Users/anastasialinchik/Desktop/Thesis/DATA/generated/human_ptms.tsv", header = T, sep = '\t')
human_ptms_hl <- read.csv(file = "/Users/anastasialinchik/Desktop/Thesis/DATA/generated/human_ptms_hl.tsv", header = T, sep = '\t')


human_ptms_expanded_hl <- read.csv(file = "/Users/anastasialinchik/Desktop/Thesis/DATA/generated/human_ptms_expanded_hl.tsv", header = T, sep = '\t')
human_ptms_expanded_length <- read.csv(file = "/Users/anastasialinchik/Desktop/Thesis/DATA/generated/human_ptms_expanded_length.tsv", header = T, sep = '\t')

hl <- read.csv(file = '/Users/anastasialinchik/Desktop/Thesis/DATA/generated/hl.tsv', header = T, sep = '\t')
cell_types_hl <- read.csv(file = '/Users/anastasialinchik/Desktop/Thesis/DATA/generated/long_proteins_hl_cell_types.tsv', header = T, sep = '\t')
long_proteins_hl <- read.csv(file = '/Users/anastasialinchik/Desktop/Thesis/DATA/generated/long_proteins_hl_cell_types.tsv', header = T, sep = '\t')
ageing_proteins <- read_excel( '/Users/anastasialinchik/Desktop/Thesis/DATA/Age-correlated protein.xlsx')
oxPTMs <- read.table(file = '/Users/anastasialinchik/Desktop/Thesis/Master_thesis/oxPTMs_unimod.txt', header = T)
```

# Normalisation

The PSM (protein-spectrum match) counts were normalised using the number of raw files that were analysed for each project. The number of raw files per project can be found on PRIDE. For some projects, it wasn't possible to analyse all of the files therefore the number of raw files was smaller than expected.

Normalisation: \`norm_counts = PSM_counts/raw_files_counts\`

# Hypotheses

## PTMs and protein half-lives

-   The half-life of a protein is the time it takes for the concentration of a radioactively protein to decrease by a half. Protein half-lives can be used as estimates of residence time of proteins in the cell.

-   Proteins that reside longer in the cell may be more susceptible to oxidative damage.

-   Protein half-lives were obtained from: [paper1](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5975921/), [paper2](https://www-science-org.kuleuven.e-bronnen.be/doi/10.1126/science.1199784#appendix).

-   It is assumed that each protein only has one modification. The proteins with no modifications are not identified.

### Distribution of protein half-lives

Approach:

-   Get a list of PTMs that correlate with ageing such as oxPTMs, acetylation etc.

-   Test whether the abundance of these PTMs changes between the long-lived proteins and the normal proteins.

The distribution of protein half-lives in the dataset can be seen here:

```{r}
#| echo: false
ggplot(human_ptms_expanded_hl, aes(x = mean_hl_hours, weight = norm_counts)) +
  geom_histogram(bins = 50, fill = "#5DB7B1") + 
  labs(title = 'Distribution of half-lives among proteins with PTMs', x = 'Mean half-lives (hours)', y = 'Normalised counts') +
  theme_classic()
```

The above distribution shows that the mean half-life of a protein in this dataset is 8.37 hours, which is within the expected range when compared to values found in [literature](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3788602/).

```{r}
#| echo: false
#| include: false

w_mean <- weighted.mean(human_ptms_hl$mean_hl_hours, human_ptms_hl$norm_counts)
w_mean
```

A boxplot was plotted in order to better visualise the proteins with exceptionally high half-lives. The boxplot does not use normalised counts due to the high number of data points. It can be seen that there are outliers in the boxplot, most of which have a high mean half-life. In total, there are 84083 outliers.

```{r}
#| echo: false
ggplot(human_ptms_hl, aes(y = mean_hl_hours)) +
  geom_boxplot() +
  theme_classic()
```

-   The distribution of the half-lives of the outliers is plotted below. Here, normalised counts are used.

```{r}
#| echo: false

# here the weights are not taken into account when calculating outliers

# Calculate boxplot statistics
boxplot_stats <- boxplot.stats(human_ptms_expanded_hl$mean_hl_hours)

# Calculate IQR
IQR <- boxplot_stats$stats[4] - boxplot_stats$stats[2]

# Define lower and upper bounds for outliers
lower_bound <- boxplot_stats$stats[2] - 1.5 * IQR
upper_bound <- boxplot_stats$stats[4] + 1.5 * IQR

# Filter outliers
outliers_df <- human_ptms_expanded_hl %>%
  filter(mean_hl_hours < lower_bound | mean_hl_hours > upper_bound)

ggplot(outliers_df, aes(y = mean_hl_hours, weight = norm_counts)) +
  geom_boxplot(fill = "#5DB7B1") + 
  labs(title = 'Distribution of half-lives among outlier proteins', y = 'Mean half-lives (hours)') +
  theme_classic()
```

Though the mean half-life of these proteins is higher than the mean half-life of the whole dataset, none of the outlier proteins can be classified as long-lived proteins. Proteins can be classified as long-lived when their mean half-life exceeds 48 hours ([ref](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5500981/#:~:text=Many%20of%20your%20body's%20proteins,%2Dlived%20protein%20(LLP).)), though this is an arbitrary definition. Nonetheless, the change in PTM abundances can still be investigated.

Violin plot for oxPTMs:

```{r}
#| echo: false
human_ptms_oxPTMs <- human_ptms_hl %>% filter(unimod_id %in% oxPTMs$ID)

ggplot(human_ptms_oxPTMs, aes(x = '', y = mean_hl_hours, weight = norm_counts)) +
  geom_violin(bw=0.5, fill = '#73BFB8') +
  labs(x = 'oxPTMs', y = 'Half-life (hours)', title = 'Distribution of protein half-lives separated by PTMs') +
  geom_abline(intercept = 2.3, slope = 0, colour = 'red') +
  theme_classic(base_size = 12)
```

-   A similar distribution is seen as before. Two peaks can be clearly seen here. It would be interesting to look at the difference between the proteins present in the two peaks. The red line is used to split the proteins into two groups.
-   Is the small peak due to the distribution of the half-lives found in various papers? This can be checked using the plot below.
-   The plot below has only one peak, meaning that small peak previously seen in the violin plot is not due to the distribution of the original half-lives.

```{r}
#| echo: false
ggplot(hl, aes(mean_hl_hours)) +
  geom_histogram(bins = 40) +
  labs(x = 'Mean half-life (hours)', y = 'Counts', title = 'Distribution of the half-lives found in papers') +
  theme_classic()
```

Below are the proteins that are present in the first peak in the violin plot.

```{r}
#| echo: false
#| eval: true
short_hl_proteins <- human_ptms_hl %>% filter(mean_hl_hours < 2.3) %>% dplyr::select(LeadProt) %>%
  distinct()

kable(short_hl_proteins)
```

ANOVA:

The group sizes are so large (there are 1 million entries between the 5 groups) that even a small difference between the groups will result in significant p-values.

```{r}
#| echo: false
#| eval: false

human_ptms_expanded_hl_top5$ptm_name <- factor(human_ptms_expanded_hl_top5$ptm_name)
# ANOVA
anova_result <- aov(mean_hl_hours ~ ptm_name, data = human_ptms_expanded_hl_top5)
summary(anova_result)

# Post-hoc tests
posthoc_tests <- glht(anova_result, linfct = mcp(ptm_name = "Tukey"))
summary(posthoc_tests)

# Checking ANOVA assumptions

# 1. Independence: Assumed to be met during experimental design.

# 2. Normality of residuals
residuals <- residuals(anova_result)
hist(residuals, main = "Histogram of Residuals")
qqnorm(residuals)
qqline(residuals)

# Shapiro-Wilk test for normality
shapiro.test(residuals)

# 3. Homogeneity of variances
plot(anova_result, 1)  # 1 corresponds to residuals vs fitted values
# Levene's test for homogeneity of variances
leveneTest(anova_result)

# 4. Linearity: Assumed to be met during model specification.
```

### Phosphorylation

-   Only the modification \[21\]Phospho is present here.

```{r}
#| echo: false
human_ptms_expanded_hl_phospho <- human_ptms_hl %>% filter(ptm_name == '[21]Phospho')

ggplot(human_ptms_expanded_hl_phospho, aes(x = mean_hl_hours, weight = norm_counts)) +
  geom_histogram(bins = 50) + 
  labs(title = 'Distribution of half lives of phosphorylated proteins', x = 'Mean half-life (hours)', y = 'Counts')+
  theme_classic()
```

Splitting the dataset in a group with phosphorylation proteins and another group with all remaining proteins.

```{r}
human_ptms_hl_mod_group <- human_ptms_hl
human_ptms_hl_mod_group$mod_group <- if_else(human_ptms_hl$unimod_id == 21, 'Phospho', 'Non-Phospho')


ggplot(human_ptms_hl_mod_group, aes(x = mean_hl_hours, weight = norm_counts, fill = mod_group)) +
  geom_density(alpha = 0.6, bw = 0.4) + 
  labs(title = 'Distribution of half lives of phosphorylated proteins', x = 'Mean half-life (hours)', y = 'Counts')+
  theme_classic()
```

### Methylation

-   Filtered by the \[34\]Methyl modification

```{r}
#| echo: false
human_ptms_expanded_hl_phospho <- human_ptms_hl %>% filter(ptm_name == '[34]Methyl')
ggplot(human_ptms_expanded_hl_phospho, aes(x = mean_hl_hours, weight = norm_counts)) +
  geom_histogram(bins = 50) + 
  labs(title = 'Distribution of half lives of phosphorylated proteins', x = 'Mean half-life (hours)', y = 'Counts')+
  theme_classic()
```

```{r}
human_ptms_hl_mod_group$mod_group <- if_else(human_ptms_hl$unimod_id == 34, 'Methyl', 'Non-Methyl')

ggplot(human_ptms_hl_mod_group, aes(x = mean_hl_hours, weight = norm_counts, fill = mod_group)) +
  geom_density(alpha = 0.6, bw = 0.4) + 
  labs(title = 'Distribution of half lives of phosphorylated proteins', x = 'Mean half-life (hours)', y = 'Counts')+
  theme_classic()
```

### Acetylation

-   Filtered by the \[1\]Acetyl modification

```{r}
#| echo: false
human_ptms_expanded_hl_phospho <- human_ptms_hl %>% filter(ptm_name == '[1]Acetyl')
ggplot(human_ptms_expanded_hl_phospho, aes(x = mean_hl_hours, weight = norm_counts)) +
  geom_histogram(bins = 50) + 
  labs(title = 'Distribution of half lives of phosphorylated proteins', x = 'Mean half-life (hours)', y = 'Counts')+
  theme_classic()
```

```{r}
#| echo: false
human_ptms_hl_mod_group$mod_group <- if_else(human_ptms_hl$unimod_id == 1, 'Acetyl', 'Non-Acetyl')

ggplot(human_ptms_hl_mod_group, aes(x = mean_hl_hours, weight = norm_counts, fill = mod_group)) +
  geom_density(alpha = 0.6, bw = 0.4) + 
  labs(title = 'Distribution of half lives of phosphorylated proteins', x = 'Mean half-life (hours)', y = 'Counts')+
  theme_classic()
```

### Oxidation

-   Only modification: \[35\]Oxidation

```{r}
#| echo: false
human_ptms_expanded_hl_oxi <- human_ptms_hl %>% filter(ptm_name == '[35]Oxidation')
ggplot(human_ptms_expanded_hl_oxi, aes(x = mean_hl_hours, weight = norm_counts)) +
  geom_histogram(bins = 50) +
  labs(title = 'Distribution of half-lives of oxidised proteins', x = 'Mean half-life (hours)', y = 'Counts')+
  theme_classic()
```

```{r}
#| echo: false

human_ptms_hl_mod_group$mod_group <- if_else(human_ptms_hl$unimod_id == 35, 'Oxidation', 'Non-Oxidation')

ggplot(human_ptms_hl_mod_group, aes(x = mean_hl_hours, weight = norm_counts, fill = mod_group)) +
  geom_density(alpha = 0.6, bw = 0.4) + 
  labs(title = 'Distribution of half lives of oxidised proteins', x = 'Mean half-life (hours)', y = 'Counts')+
  theme_classic()
```

### oxPTMs

All PTMs related to oxidative damage in general, not only \[35\]Oxidation.

```{r}
#| echo: false
human_ptms_hl_oxi <- human_ptms_hl %>% filter(unimod_id %in% oxPTMs$ID)
ggplot(human_ptms_hl_oxi, aes(x = mean_hl_hours, weight = norm_counts)) +
  geom_histogram(bins = 50) +
  labs(title = 'Distribution of half-lives of oxidised proteins', x = 'Mean half-life (hours)', y = 'Counts')+
  theme_classic()
```

```{r}
#| echo: false
human_ptms_hl_mod_group$mod_group <- if_else(human_ptms_hl$unimod_id %in% oxPTMs$ID, 'oxPTMs', 'Non-oxPTMs')

ggplot(human_ptms_hl_mod_group, aes(x = mean_hl_hours, weight = norm_counts, fill = mod_group)) +
  geom_density(alpha = 0.5, bw = 0.4) +
  labs(title = 'Distribution of half-lives of oxidised proteins', x = 'Mean half-life (hours)', y = 'Counts')+
  theme_classic()
```

```{r}
#| echo: false

df <- human_ptms_hl_mod_group %>% dplyr::select(LeadProt, modification, mod_group, unimod_id, mean_hl_hours) %>%
  distinct()

ggplot(df, aes(x = mod_group, y = mean_hl_hours)) +
  geom_violin(bw=0.4, fill = '#73BFB8') +
  geom_boxplot(width=0.3, outlier.shape = NA) + 
  labs(x = 'PTM name', y = 'Half-life (hours)', title = 'Distribution of protein half-lives separated by PTMs') +
  stat_summary(fun=mean, geom="point", shape=19, size=2.5, colour = '#EA7317') +
  theme_classic(base_size = 12)
```

As can be seen in the above graphs, the presence of the small peak, which represents proteins with very short half-lives, varies depending on the type of modification. It is more prominent in PTMs that are related to ageing.

**Hypothesis**: proteins with higher mean half-lives remain in the cell for longer, therefore they are more susceptible to oxidative damage and will accumulate more oxPTMs.

Approach:

-   Count the total number of oxPTMs on each protein (using normalised counts)

-   Plot the mean half-life of the proteins vs their total oxPTMs count

Below is a plot that shows how the number of oxPTMs changes with the mean half-life of different proteins. The proteins, which were identified to have a very short half-life are shown in blue. A similar pattern is seen between the blue and red points, where some have a very high number of modifications, while most show very low abundances. Most proteins still seem to remain largely unmodified irrespective of their half-life.

```{r}
#| echo: false
#| warning: false
oxPTMs_hl <- human_ptms_hl %>% 
  inner_join(oxPTMs, by = join_by(unimod_id == ID)) %>%
  group_by(LeadProt) %>%
  summarise(mean_hl_hours, sum_oxPTMs = sum(norm_counts)) %>%
  distinct()

oxPTMs_hl$group <- if_else(oxPTMs_hl$LeadProt %in% short_hl_proteins$LeadProt, 'short hl', 'normal hl') #SHL = short half-life, NHL = normal half-life
ggplot(oxPTMs_hl, aes(x = mean_hl_hours, y = sum_oxPTMs, colour = group)) +
  geom_point() +
  theme_classic()
```

## PTMs in long-lived proteins

Long-lived proteins can be used as estimators of chronological age. Long-lived proteins can be defined in different ways, for example based on the half-life of the protein when compared to the average half-life of proteins in the organism. In this case, long-lived proteins were obtained from the following study: [paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3788602/). Proteins were classified as long-lived based on their degree of degradation during the experiment and therefore it was possible to discover new long-lived proteins (no *a priori* assumptions were made).

The study identified a list of long-lived proteins in rats, therefore human orthologs of these proteins were found.

```{r}
#|echo: false

long_lived_proteins <- read_excel(path = '/Users/anastasialinchik/Desktop/Thesis/DATA/long_lived_proteins.xlsx') %>% subset(select = c(3,4))
long_lived_proteins <- na.omit(long_lived_proteins)
long_lived_proteins_hl <- inner_join(human_ptms, long_lived_proteins, by = join_by(LeadProt == LeadProt))

human_ptms_hl_long_proteins <- rbind(long_lived_proteins_hl, human_ptms_hl)

human_ptms_hl_long_proteins$group <- if_else(human_ptms_hl_long_proteins$LeadProt %in% long_lived_proteins$LeadProt, 'long lived', 'non-long lived')
```

The plot below shows the distribution of the half-lives of proteins with oxPTMs (normalised counts were used). Clearly there is a much greater number of short lived proteins compared to long-lived proteins.

```{r}
#|echo: false
human_ptms_hl_oxi <- human_ptms_hl_long_proteins %>% filter(unimod_id %in% oxPTMs$ID)

ggplot(human_ptms_hl_oxi, aes(x = mean_hl_hours, weight = norm_counts)) +
  geom_histogram(bins = 50) +
  labs(title = 'Distribution of half-lives of oxidised proteins', x = 'Mean half-life (hours)', y = 'Counts')+
  theme_classic()
```

### oxPTMs

This plot is similar to the one already seen above, however this time, long-lived proteins are included. There is still no correlation between the abundance of PTMs and protein half-lives.

```{r}
#| echo: false
#| warning: false
oxPTMs_hl <- human_ptms_hl_oxi %>%
  group_by(LeadProt) %>%
  summarise(group, mean_hl_hours, sum_oxPTMs = sum(norm_counts)) %>%
  distinct()


ggplot(oxPTMs_hl, aes(x = mean_hl_hours, y = sum_oxPTMs, colour = group)) +
  geom_point() +
  theme_classic()
```

## PTMs and protein lengths

Protein lengths were found on UniProt.

The assumption is made that each protein can only have 1 PTM.

```{r}
#| echo: false

ggplot(human_ptms_expanded_length, aes(x = Length, weight = norm_counts)) +
  geom_histogram(fill = '#73BFB8', bins = 50) +
  labs(title = 'Distribution of protein lengths', x = 'Length (in amino acids)', y = 'Normalised counts') +
  theme_classic(base_size = 13)
```

Proteins that are longer than 20,000 amino acids.

```{r}
#| echo: false
long_proteins <- human_ptms_expanded_length %>% filter(Length > 20000) %>% dplyr::select(Uniprot_entry_name, Length) %>% distinct()

kable(long_proteins)
```

```{r}
#| echo: false
#| warning: false
ggplot(human_ptms_expanded_length, aes(x = Length, weight = norm_counts)) +
  geom_histogram(fill = '#73BFB8', bins = 50) +
  labs(title = 'Distribution of protein lengths with less than 10,000 amino acids', x = 'Length (in amino acids)', y = 'Counts') +
  scale_x_continuous(limits=c(0, 10000))+
  theme_classic(base_size = 13)
```

```{r}
#| echo: false
human_ptms_expanded_length %>% 
  group_by(LeadProt) %>%
  mutate(PTMs_per_protein = sum(norm_counts)) %>%
  dplyr::select(LeadProt, Length, PTMs_per_protein) %>% distinct() %>%
  ggplot(aes(x = Length, y = PTMs_per_protein)) +
  geom_point() + 
  labs(title = 'Number of PTMs in function of protein length', x = 'Protein length (amino acids)', y = 'Number of PTMs (normalised)') +
  theme_classic()


```

```{r}
#| echo: false
human_ptms_expanded_length %>% 
  group_by(LeadProt) %>%
  mutate(PTMs_per_protein = sum(norm_counts)) %>%
  dplyr::select(LeadProt, Length, PTMs_per_protein) %>% distinct() %>% 
  filter(Length < 10000) %>%
  ggplot(aes(x = Length, y = PTMs_per_protein)) +
  geom_point() +
  labs(title = 'Number of PTMs in function of protein length (zoomed in)', x = 'Protein length (amino acids)', y = 'Number of PTMs (normalised)') +
  theme_classic()
```

There seems to be no correlation between the length of the protein and oxPTMs.

## PTMs and melting temperatures

```{r}
#| echo: false
mt <- read_xlsx(path = '/Users/anastasialinchik/Desktop/Thesis/DATA/MeltingTemperatures.xlsx', sheet = 5,  col_names = T)

mt <- separate_rows(mt, Protein_ID, sep = ";")
mt <- mt %>% dplyr::select(Protein_ID, `Tm Protein`) %>% group_by(Protein_ID) %>% summarise(mean_mt = mean(`Tm Protein`)) 
```

Below is the distribution of the melting temperatures found in literature.

```{r}
#| echo: false
ggplot(mt, aes(mean_mt)) +
  geom_histogram() +
  theme_classic()
```

Below is a plot of the distribution of the melting temperatures in the dataset with PTMs.

```{r}
#| echo: false
ptms_mt <- inner_join(human_ptms, mt, by = join_by(LeadProt == Protein_ID))
ggplot(ptms_mt, aes(x = mean_mt, weight = norm_counts)) +
  geom_histogram() +
  theme_classic()

```

Below is a plot the mean half-lives and the weighted abundance of PTMs (specific PTMs were not selected). There seems to be no correlation between the mean melting temperature of the proteins and the number of modifications.

```{r}
ptms_mt_all <- ptms_mt %>% group_by(LeadProt) %>%
  mutate(sum_norm_counts = sum(norm_counts)) %>% 
  dplyr::select(LeadProt, sum_norm_counts, mean_mt, unimod_id) %>%
  distinct()
ggplot(ptms_mt_all, aes(x = mean_mt, y = sum_norm_counts)) +
  geom_point() +
  theme_classic(base_size = 13)
```

### Only oxPTMs

```{r}
ptms_mt_oxPTMs <- ptms_mt %>% 
  filter(unimod_id %in% oxPTMs$ID) %>%
  group_by(LeadProt) %>%
  mutate(sum_norm_counts = sum(norm_counts)) %>% 
  dplyr::select(LeadProt, sum_norm_counts, mean_mt, unimod_id) %>%
  distinct()

ggplot(ptms_mt_oxPTMs, aes(x = mean_mt, y = sum_norm_counts)) +
  geom_point() +
  theme_classic(base_size = 13)
```

Only oxidation is counted in the plot below:

```{r}
ptms_mt %>% 
  filter(unimod_id == 35) %>%
  group_by(LeadProt) %>%
  mutate(sum_norm_counts = sum(norm_counts)) %>% 
  dplyr::select(LeadProt, sum_norm_counts, mean_mt, unimod_id) %>%
  distinct() %>%
  ggplot(aes(x = mean_mt, y = sum_norm_counts)) +
  geom_point() +
  theme_classic(base_size = 13)
```

## PTMs and disorder

IUPred REST API was used to get all of the IUPred scores for all of the proteins. IUPred outputs a score for each amino acid within a protein. A Python script was used to get the scores for each amino acid in a protein and average them for each protein; the resulting score is taken as the IUPred score for that protein. Below is the distribution of the IUPred scores in the dataset.

```{r}
#| echo: false
iupred_scores <- read.csv(file = "/Users/anastasialinchik/Desktop/Thesis/DATA/generated/iupred_scores.tsv", header = T, sep = '\t')

human_ptms_iupred <- inner_join(human_ptms, iupred_scores, by = join_by(LeadProt == UniProt_ID)) 

ggplot(iupred_scores, aes(Mean_IUPRED_Score)) +
  geom_histogram() +
  theme_classic()
```

The plot below shows how the weighted abundance of modifications changes with the IUPred score. There is no correlation between the two variables.

```{r}
#| echo: false
#| warning: false
human_ptms_iupred <- human_ptms_iupred %>% group_by(LeadProt) %>%
  summarise(LeadProt, Mean_IUPRED_Score, sum_norm_counts = sum(norm_counts)) 

ggplot(human_ptms_iupred, aes(x = Mean_IUPRED_Score, y = sum_norm_counts)) +
  geom_point() +
  theme_classic()
```

There is also an R package that can be used to calculate the IUPred score, however it uses the old version of IUPred and it also takes a very long time to load. (Though the Python script also took \~ 7 hours to execute)

[IUPred Vignette (bioconductor.org)](https://bioconductor.org/packages/release/bioc/vignettes/idpr/inst/doc/iupred-vignette.html)

```{r}
#| eval: false
#| echo: false
# this uses the R package for IUPRED2. The REST API was used bcs it uses the latest version of IUPRED so more proteins are recognised + it is faster. 
#uniprot_ids <- c('P01023', 'P42684', 'Q9BYF1')

predict_disorder_scores <- function(uniprot_ids) {
  disorder_scores <- c()
  
  for (id in uniprot_ids) {
    disorder_df <- iupred(
      id,
      iupredType = "long",
      plotResults = F,
      proteinName = NA
    )
    
    disorder_score <-
      disorder_df %>% summarise(mean_score = mean(IUPred2))
    disorder_scores <- append(disorder_scores, disorder_score[1, 1])
    
  }
  
  disorder_df <-
    data.frame(uniprot_id = uniprot_ids, disorder_score = disorder_scores)
  
  return(disorder_df)
}

missing_uniprot_ids <- c('A0A024R7F8', 'A0A024R127')
ageing_proteins_ids_disorder <- ageing_proteins_ids %>% filter(! `Uniprot entry` %in% missing_uniprot_ids)

disorder_df <- predict_disorder_scores(ageing_proteins_ids_disorder$`Uniprot entry`)
```

## Chaperones

Chaperones get modified through their life: [Regulation of molecular chaperones through post-translational modifications: Decrypting the chaperone code - PMC (nih.gov)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4492711/#:~:text=The%20post%2Dtranslational%20modification%20(PTM)%20of%20molecular%20chaperones%2C,activity%20of%20these%20proteins%20that)

The list of chaperones that are repressed or induced during ageing was obtained [here](https://www.sciencedirect.com/science/article/pii/S2211124714008250?via%3Dihub#app2).

Ideas:

-   Chaperone code

    -   does phosphorylation change in chaperones due to ageing
    -   take collagen as a reference protein and use the number of modifications in collagen in a particular project as an indication of age. Then look at the correlation between collagen modifications and modifications in chaperones.
        -   will be biased though

```{r}
#| echo: false
#| include: false
repressed_chaperones <- read_excel(path = '/Users/anastasialinchik/Desktop/Thesis/DATA/subnetwork_chaperones.xlsx', sheet = 'BREHME_TABLE S3D', skip = 19)

repressed_chaperones <- separate_wider_delim(repressed_chaperones, col = 'Label [Family]', names = c("gene_name", "family"), delim = " ")


induced_chaperones <- read_excel(path = '/Users/anastasialinchik/Desktop/Thesis/DATA/subnetwork_chaperones.xlsx', sheet = 'BREHME_TABLE S3E', skip = 19)

induced_chaperones <- separate_wider_delim(induced_chaperones, col = 'Label [Family]', names = c("gene_name", "family"), delim = " ")
```

```{r}
#| echo: false
chaperones <- read.csv(file = "/Users/anastasialinchik/Desktop/Thesis/idmapping_2024_03_17.tsv", sep = '\t', header = T)
```

### What modifications are present on the chaperones

The table below shows all of the modifications that are present on chaperones. These are chaperones that were found to decrease or increase in abundance during ageing. It can be seen that some chaperones are susceptible to more modifications compared to others.

```{r}
#| echo: false
chaperone_ptms <- inner_join(human_ptms, chaperones, by = join_by(LeadProt == Entry))

chaperone_ptms_grouped <- chaperone_ptms %>%
  group_by(Uniprot_entry_name) %>%
  summarize(PTMs = paste(unique(ptm_name), collapse = "; "))

table <- kable(chaperone_ptms_grouped, 
      caption = "Summary of PTMs by PTM Residue",
      align = "l",              # Align all columns to the left
      col.names = c("PTM Residue", "Post-translational modification"),  # Specify column names
      format = "html",          # Format the table as HTML
      escape = FALSE,           # Allow HTML formatting
      table.attr = "style='width:50%; margin-left:auto; margin-right:auto;'", # Center the table and set width
      row.names = FALSE,        # Do not display row names
      bold.headers = TRUE      # Make headers bold
)
table

# chaperone_ptms_grouped %>% kbl() %>% kable_styling()
```
